include/seqan3/argument_parser/detail/version_check.hpp

```diff
@@ -0,0 +1,415 @@
+
+            std::getline(version_file, line); // get second line which should only contain the version number of seqan
+            srv_seqan_version = get_numbers_from_version_string(line);
+
+            version_file.close();
+        }
+
+#if !defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_) // only check seqan version in debug or testing mode
+        if (srv_seqan_version != empty_version)
+        {
+            std::array<int, 3> seqan_version = {SEQAN3_VERSION_MAJOR, SEQAN3_VERSION_MINOR, SEQAN3_VERSION_PATCH};
+
+            if (seqan_version < srv_seqan_version)
+                std::cerr << MESSAGE_SEQAN3_UPDATE;
+        }
+#endif
+
+        if (srv_app_version != empty_version) // app version
+        {
+#if defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_) // only check app version in release or testing mode
+            if (get_numbers_from_version_string(version) < srv_app_version)
+                std::cerr << MESSAGE_APP_UPDATE;
+#endif // defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_)
+
+#if !defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_) // only notify developer that app version should be updated on server
+            if (get_numbers_from_version_string(version) > srv_app_version)
+                std::cerr << MESSAGE_REGISTERED_APP_UPDATE;
+#endif // !defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_)
+        }
+
+        std::cerr << std::flush;
+
+        if (program.empty())
+        {
+            prom.set_value(false);
+            return;
+        }
+
+        // launch a separate thread to not defer runtime.
+        if (cookie_line == "ALWAYS")
+            std::thread(call_server, "", command, std::move(prom)).detach();
+        else
+            std::thread(call_server, timestamp_filename, command, std::move(prom)).detach();
+
+    }
+
+    //!\brief The identification string that may appear in the version file if an app is unregistered.
+    static constexpr char const * const UNREGISTERED_APP = "UNREGISTERED_APP";
+    //!\brief The message directed to the developer of the app if a new seqan3 version is available.
+    static constexpr char const * const MESSAGE_SEQAN3_UPDATE =
+        "[SEQAN3 INFO] :: A new SeqAn3 version is available online.\n"
+        "[SEQAN3 INFO] :: Please visit www.github.com/seqan/seqan3.git for an update\n"
+        "[SEQAN3 INFO] :: or inform the developer of this app.\n"
+        "[SEQAN3 INFO] :: If you don't wish to receive further notifications, set --version-check OFF.\n\n";
+    //!\brief The message directed to the developer of the app if the app is not yet registered with us.
+    static constexpr char const * const MESSAGE_UNREGISTERED_APP =
+        "[SEQAN3 INFO] :: Thank you for using SeqAn!\n"
+        "[SEQAN3 INFO] :: Do you wish to register your app for update notifications?\n"
+        "[SEQAN3 INFO] :: Just send an email to support@seqan.de with your app name and version number.\n"
```

Comment:
Here is a different e-mail address now. They should definitely be kept in sync with the doc.